pca9685: add buffered one shot write

soypat opened the pull request:
This adds functionality for fast peripheral writes needed for control of real-time systems such as drones.

soypat committed:
pca9685: add buffered one shot write

sago35 commented on pca9685/pca9685_buffered.go:
```diff
@@ -0,0 +1,48 @@
+// PrepPhasedSet prepares a phased PWM value to be written to the
+// channel's register on Update() call.
+func (b *DevBuffered) PrepPhasedSet(channel uint8, on, off uint32) {
+	onLReg := channel * 4
+	binary.LittleEndian.PutUint16(b.ledBuf[onLReg+1:], uint16(on)&maxtop)
+	binary.LittleEndian.PutUint16(b.ledBuf[onLReg+3:], uint16(off)&maxtop)
```
I think it would be easier to understand if accessed in the following way.


```suggestion
	onLReg := 1 + channel * 4
	binary.LittleEndian.PutUint16(b.ledBuf[onLReg:], uint16(on)&maxtop)
	binary.LittleEndian.PutUint16(b.ledBuf[onLReg+2:], uint16(off)&maxtop)
```

This is because the buf configuration is as follows.

* [0] : LEDSTART
* [1:5] : channel 0
* [5:9] : channel 1
* ...
* [1 + N\*4 : 1 + N\*4 + 4] : channnel N

sago35 reviewed (commented):
@soypat 
Please add examples for this change.
Please also add to the Makefile.

I think it is good except for the part I commented on.

soypat commented:
@sago35 Thank you for the feedback. I have added documentation to the ledBuf field and corrected the part you commented on. Let me know of anything else!

deadprogram commented:
Looks good, thanks @soypat now merging.
